File: csharp14/Common/MenuChoice.cs
namespace DirectorySize.Common;

public class MenuChoice(string path, string label)
{
    public static readonly MenuChoice Up = new("up", Utils.EscapeMarkup("[[..] Up]"));
    public static readonly MenuChoice Exit = new("exit", Utils.EscapeMarkup("[Exit]"));

    public string Path { get; } = path;
    public string Label { get; } = label;

    public override string ToString() => Label;
}

File: fsharp/Common/MenuChoice.fs
namespace DirectorySize.Common

open DirectorySize.Common

type MenuChoice(path: string, label: string) =
    static member Up = MenuChoice("up", Utils.escapeMarkup("[[..] Up]"))
    static member Exit = MenuChoice("exit", Utils.escapeMarkup("[Exit]"))

    member _.Path = path
    member _.Label = label

    override _.ToString() = label
